apps/emqx_gateway/src/coap/emqx_coap_api.erl

```diff
@@ -131,15 +127,22 @@ parse_payload(_, Body) ->
     Body.
 
 call_client(ClientId, Msg, Timeout) ->
-    case emqx_gateway_cm_registry:lookup_channels(coap, ClientId) of
-        [Channel | _] ->
-            RequestId = emqx_coap_channel:send_request(Channel, Msg),
-            case gen_server:wait_response(RequestId, Timeout) of
-             {reply, Reply} ->
-                 Reply;
-             _ ->
-                 timeout
-            end;
-        _ ->
+    try
+        case emqx_gateway_cm_registry:lookup_channels(coap, ClientId) of
+            [Channel | _] ->
+                RequestId = emqx_coap_channel:send_request(Channel, Msg),
+                case gen_server:wait_response(RequestId, Timeout) of
+                    {reply, Reply} ->
+                        Reply;
+                    _ ->
+                        timeout
+                end;
+            _ ->
+                not_found
+        end
+    catch _:Error ->
+            ?SLOG(warning, #{msg => "invalid coap lookup",
+                             client_id => ClientId,
+                             error => Error}),
```

Comment:
Perhaps we could print out the stacktrace, which would be more useful for analyzing the error from the log?